newIDE/app/src/UI/Theme/FullThemeProvider.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+import { ThemeProvider } from '@material-ui/core';
+import { StylesProvider, jssPreset } from '@material-ui/core/styles';
+import * as React from 'react';
+import { getFullTheme } from '.';
+import { create } from 'jss';
+import rtl from 'jss-rtl';
+import GDevelopThemeContext from './GDevelopThemeContext';
+import PreferencesContext from '../../MainFrame/Preferences/PreferencesContext';
+import { useResponsiveWindowWidth } from '../Reponsive/ResponsiveWindowMeasurer';
+
+// Add the rtl plugin to the JSS instance to support RTL languages in material-ui components.
+const jss = create({
+  plugins: [...jssPreset().plugins, rtl()],
+});
+
+type Props = {|
+  children: React.Node,
+  forcedThemeName?: string,
+|};
+
+export const FullThemeProvider = ({ children, forcedThemeName }: Props) => {
```

Comment:
Nice to have everything here!

Reply:
Easier to read, and a single "source of truth" for the theme 👍